tests/ballerina-tools-integration-test/src/test/java/org/ballerinalang/test/packaging/PackagingTestCase.java

```diff
@@ -204,6 +201,48 @@ public void testInstall() throws Exception {
         Assert.assertTrue(Files.exists(tempHomeDirectory.resolve(dirPath).resolve(packageName + ".zip")));
     }
 
+    @Test(description = "Test pulling a package from central", dependsOnMethods = "testPush")
+    public void testPull() throws Exception {
+        Path dirPath = Paths.get(ProjectDirConstants.CACHES_DIR_NAME,
+                                 ProjectDirConstants.BALLERINA_CENTRAL_DIR_NAME,
+                                 "integrationtests", packageName, "1.0.0");
+
+        given().with().pollInterval(Duration.TEN_SECONDS).and()
+               .with().pollDelay(Duration.FIVE_SECONDS)
+               .await().atMost(60, SECONDS).until(() -> {
+            ballerinaClient = new ServerInstance(serverZipPath);
+            String[] clientArgs = {"integrationtests/" + packageName + ":1.0.0"};
+            ballerinaClient.runMain(clientArgs, getEnvVariables(), "pull");
+            return Files.exists(tempHomeDirectory.resolve(dirPath).resolve(packageName + ".zip"));
+        });
+
+        Assert.assertTrue(Files.exists(tempHomeDirectory.resolve(dirPath).resolve(packageName + ".zip")));
+    }
+
+    @Test(description = "Test searching a package from central", dependsOnMethods = "testPush")
+    public void testSearch() throws BallerinaTestException, IOException {
+        ballerinaClient = new ServerInstance(serverZipPath);
+        String[] clientArgs = {packageName};
+
+        DateTimeFormatter dtf = DateTimeFormatter.ofPattern("yyyy-MM-dd-EE");
+        String date = dtf.format(LocalDateTime.now());
```

Comment:
Shall we move this to testPush() since theres a chance the date would change when running the tests.